mlflow/pytorch/pytorch_autolog.py

```diff
@@ -1,180 +1,156 @@
-                )
-                if hasattr(optimizer, "defaults"):
-                    defaults_dict = optimizer.defaults
-                    if "lr" in defaults_dict:
-                        trainer.logger.experiment.log_param(
-                            trainer.logger.run_id, "learning_rate", defaults_dict["lr"]
-                        )
-
-                    if "eps" in defaults_dict:
-                        trainer.logger.experiment.log_param(
-                            trainer.logger.run_id, "epsilon", defaults_dict["eps"]
-                        )
-
-        summary = str(ModelSummary(pl_module, mode="full"))
-        tempdir = tempfile.mkdtemp()
-        try:
-            summary_file = os.path.join(tempdir, "model_summary.txt")
-            with open(summary_file, "w") as f:
-                f.write(summary)
-
-            trainer.logger.experiment.log_artifact(
-                trainer.logger.run_id, local_path=summary_file
-            )
-        finally:
-            shutil.rmtree(tempdir)
-
-    def on_train_end(self, trainer, pl_module):
-        """
-        Logs the model checkpoint into mlflow - models folder on the training end
-        """
+        def __init__(self):
+            super().__init__()
+
+        def on_epoch_end(self, trainer, pl_module):
+            """
+            Log loss and other metrics values after each epoch
+            """
+            if (pl_module.current_epoch - 1) % every_n_iter == 0:
+                for key, value in trainer.callback_metrics.items():
+                    try_mlflow_log(mlflow.log_metric, key, float(value), step=pl_module.current_epoch)
+
+            if trainer.early_stop_callback:
+                self._early_stop_check(trainer.early_stop_callback)
+
+        def on_train_start(self, trainer, pl_module):
+            """
+            Logs Optimizer related metrics when the train begins
+            """
+            mlflow.set_tag(key="Mode", value="training")
+            if trainer.early_stop_callback:
+                self._log_early_stop_params(trainer.early_stop_callback)
+
+            if hasattr(trainer, "optimizers"):
+                for optimizer in trainer.optimizers:
+                    try_mlflow_log(mlflow.log_param, "optimizer_name", type(optimizer).__name__)
+
+                    if hasattr(optimizer, "defaults"):
+                        defaults_dict = optimizer.defaults
+                        if "lr" in defaults_dict:
+                            try_mlflow_log(mlflow.log_param, "learning_rate", defaults_dict["lr"])
+
+                        if "eps" in defaults_dict:
+                            try_mlflow_log(mlflow.log_param, "epsilon", defaults_dict["eps"])
+
+            summary = str(ModelSummary(pl_module, mode="full"))
+            tempdir = tempfile.mkdtemp()
+            try:
+                summary_file = os.path.join(tempdir, "model_summary.txt")
+                with open(summary_file, "w") as f:
```

Comment:
What happens if this file is already pre-existing? Please check there are no adverse effects

Reply:
In this case, we are creating a new temp directory and writing the model summary into it. Finally, we are deleting the temp directory as well. Hence, there is no adverse effects here.